rskj-core/src/main/java/co/rsk/peg/PeginInformation.java

```diff
@@ -0,0 +1,131 @@
+            if (peginInstructionsOptional.isPresent()) {
+                PeginInstructions peginInstructions = peginInstructionsOptional.get();
+                parseFromPeginInstructions(peginInstructions);
+            }
+        }
+
+        // If BtcLockSender could not be parsed and peg-in instructions were not provided, then this tx can't be processed
+        if(!btcLockSenderOptional.isPresent() && !peginInstructionsOptional.isPresent()) {
+            String message = String.format("Could not get peg-in information for tx %s", btcTx.getHash());
+            logger.warn("[parse] {}", message);
+            throw new PeginInstructionsException(message);
+        }
+    }
+
+    private void parseFromBtcLockSender(BtcLockSender btcLockSender) {
+        this.protocolVersion = 0;
+        this.rskDestinationAddress = btcLockSender.getRskAddress();
+        this.btcRefundAddress = btcLockSender.getBTCAddress();
+        this.senderBtcAddress = btcLockSender.getBTCAddress();
+        this.senderBtcAddressType = btcLockSender.getTxSenderAddressType();
+
+        logger.trace("[parseFromBtcLockSender] Protocol version: {}", this.protocolVersion);
+        logger.trace("[parseFromBtcLockSender] RSK destination address: {}", btcLockSender.getRskAddress());
+        logger.trace("[parseFromBtcLockSender] BTC refund address: {}", btcLockSender.getBTCAddress());
+        logger.trace("[parseFromBtcLockSender] Sender BTC address: {}", btcLockSender.getBTCAddress());
+        logger.trace("[parseFromBtcLockSender] Sender BTC address type: {}", btcLockSender.getTxSenderAddressType());
+    }
+
+    private void parseFromPeginInstructions(PeginInstructions peginInstructions)  throws PeginInstructionsException {
+        this.protocolVersion = peginInstructions.getProtocolVersion();
+        this.rskDestinationAddress = peginInstructions.getRskDestinationAddress();
+        logger.trace("[parseFromPeginInstructions] Protocol version: {}", peginInstructions.getProtocolVersion());
+        logger.trace("[parseFromPeginInstructions] RSK destination address: {}", peginInstructions.getRskDestinationAddress());
+
+        switch (protocolVersion) {
+            case 1:
+                PeginInstructionsVersion1 peginInstructionsV1 = (PeginInstructionsVersion1) peginInstructions;
+                parseFromPeginInstructionsVersion1(peginInstructionsV1);
+                break;
+            default:
+                String message = String.format("Invalid protocol version: %d", protocolVersion);
+                logger.warn("[parseFromPeginInstructions] {}", message);
+                throw new PeginInstructionsException(message);
+        }
+    }
+
+    private void parseFromPeginInstructionsVersion1(PeginInstructionsVersion1 peginInstructions) {
+        Optional<Address> btcRefundAddressOptional = peginInstructions.getBtcRefundAddress();
+        if (btcRefundAddressOptional.isPresent()) {
```

Comment:
We might want to prevent addresses like the federation here. Not a big deal, but would just waste resources